statsd/uds_async.go

```diff
@@ -0,0 +1,119 @@
+package statsd
+
+import (
+	"fmt"
+	"net"
+	"time"
+)
+
+// asyncUdsWriter is an internal class wrapping around management of UDS connection
+type asyncUdsWriter struct {
+	// Address to send metrics to, needed to allow reconnection on error
+	addr net.Addr
+	// Established connection object, or nil if not connected yet
+	conn net.Conn
+	// write timeout
+	writeTimeout time.Duration
+	// datagramQueue is the queue of datagrams ready to be sent
+	datagramQueue chan []byte
+	stopChan      chan struct{}
+}
+
+// New returns a pointer to a new asyncUdsWriter given a socket file path as addr.
+func newAsyncUdsWriter(addr string) (*asyncUdsWriter, error) {
+	udsAddr, err := net.ResolveUnixAddr("unixgram", addr)
+	if err != nil {
+		return nil, err
+	}
+
+	writer := &asyncUdsWriter{
+		addr:         udsAddr,
+		conn:         nil,
+		writeTimeout: defaultUDSTimeout,
```

Comment:
Do we want to make this TO overridable? Currently there's no real clean interface to do that, and since these are not exported it wouldn't allow users to achieve that. Same goes for the queue depth (8192).

I believe they're both quite sensitive parameters that should be configurable.

Reply:
It will be in #82